dascore/proc/whiten.py

```diff
@@ -0,0 +1,225 @@
+"""Spectral whitening."""
+from __future__ import annotations
+
+import math
+
+import numpy as np
+import numpy.fft as nft
+import pandas as pd
+from scipy.ndimage import convolve1d
+from scipy.signal.windows import tukey
+
+from dascore.constants import PatchType
+from dascore.exceptions import ParameterError
+from dascore.utils.misc import check_filter_kwargs, check_filter_range
+from dascore.utils.patch import get_dim_sampling_rate
+
+
+def whiten(
+    patch: PatchType,
+    freq_smooth_size: float | None = None,
+    tukey_alpha: float = 0.1,
+    **kwargs,
+) -> PatchType:
+    """
+    Band-limited signal whitening.
+
+    Parameters
+    ----------
+    patch
+        Patch to transform. Has to have dimensions of time and distance.
+    freq_range
+        Frequency range for which the whitening is applied, in Hz.
+        If no value is inputted, the entire spectrum is used.
+    freq_smooth_size
+        Size (in Hz) of moving average window, used to compute the spectrum
+        before whitening. If no value is inputted, smoothing is over
+        the entire spectrum.
+    tukey_alpha
+        Alpha parameter for Tukey window applied as windowing to the
+        smoothed spectrum within the required frequency range. By default
+        its value is 0.1.
+        See more details at https://docs.scipy.org/doc/scipy/reference
+        /generated/scipy.signal.windows.tukey.html
+    **kwargs
+        Used to specify the dimension and frequency, wavelength, or equivalent
+        limits.
```

Comment:
Add info about what happens if kwargs are not specified (last axis is used, 0 to nyquist is the range right?)

Reply:
True that. fixed.